src/main/java/com/riscure/trs/parameter/trace/TraceParameterMap.java

```diff
@@ -32,55 +36,121 @@ public static TraceParameterMap deserialize(byte[] bytes, TraceParameterDefiniti
         return result;
     }
 
+    public <T> void put(TypedKey<T> typedKey, T value) {
+        put(typedKey.getKey(), TraceParameterFactory.create(typedKey.getCls(), value));
+    }
+
+    public <T> T get(TypedKey<T> typedKey) {
+        TraceParameter traceParameter = Optional.ofNullable(get(typedKey.getKey()))
+                .orElseThrow(() -> new RuntimeException(String.format(KEY_NOT_FOUND, typedKey.getKey())));
```

Comment:
how about `NoSuchElementException`?

```suggestion
                .orElseThrow(() -> new NoSuchElementException(String.format(KEY_NOT_FOUND, typedKey.getKey())));
```